numpy/core/setup.py

```diff
@@ -605,6 +606,12 @@ def generate_api(ext, build_dir):
     config.add_include_dirs(join('src', 'umath'))
     config.add_include_dirs(join('src', 'npysort'))
 
+    # Need to add location of npy_math_internal.h, generated from src/npymath/npy_math_internal.h.src
+    npymath_build_dir = join('build', 'src.%s-%s.%s' % (sysconfig.get_platform(), sys.version_info[0], sys.version_info[1]),
```

Comment:
The build director is already defined somewhere else that is accessible to the setup.py one might as well use it instead of repeating the code. I think it is even defined inside python itself so out of our control which is a even better reason to not hardcode the current status quo even when it is unlikely to change.

It is more a concern for the stuff that are actually compiler builtins like copysign. The capability checking inside header files is much more limited that what numpy can do during configuration (unfortunately gcc still does not have  __has_builtin like clang does)
It is probably of less concern nowadays with functionality used in numpy is present in compilers since decades on linux and windows is mostly binary distribution anyway. So installing inlinable headers is something we can consider in future.